lib/practice_exercises.rb

```diff
@@ -1,13 +1,43 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  # raise NotImplementedError, "Not implemented yet"
+  if string == ""
+    return true
+  end
+
+  string = string.downcase.tr('.,:;?/|~!@#$%^&*() ', '')
+
+  index = 0
+  while index <= string.length / 2 do
+    if string[index] != string[string.length - index - 1]
+      return false
+    end
+    index += 1
+  end
+  return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+string = "!@#$%^&*() racecar "
+puts "#{is_palindrome(string)}"
+
+# Time Complexity: O(n^2)
+# Space Complexity: O(n)
 def longest_prefix(strings)
```

Comment:
This works, but since you're looping through the list of strings (n) and then looping through the letters in the shortest word in the list (m).  So this would be O(m) time complexity.